encoding/test_encoding.py

```diff
@@ -0,0 +1,615 @@
+        server = self.get_server("backend")
+
+        request = [
+            (":authority", "localhost"),
+            (":path", "/"),
+            (":scheme", "https"),
+            (":method", "GET"),
+        ]
+        client.make_request(request)
+        got_response = client.wait_for_response(timeout=5)
+        response1 = client.responses[-1] if len(client.responses) else None
+
+        self.assertTrue(got_response, "Got no response")
+        self.assertFalse(
+            "Transfer-Encoding" in response1.headers,
+            "The response should not have Transfer-Encoding",
+        )
+        cl = response1.headers.get("Content-Length", None)
+        self.assertTrue(cl, "The response should have Content-Length")
+
+        # repeat the response to ensure it wasn't cached
+        client.make_request(request)
+        got_response = client.wait_for_response(timeout=5)
+        response2 = client.responses[-1] if len(client.responses) else None
+
+        self.assertTrue(got_response, "Got no response")
+        self.assertEqual(len(server.requests), 2, "The response must not be served from the cache")
+        self.assertEqual(
+            response1.headers.get("Content-Length"),
+            response2.headers.get("Content-Length"),
+            "Requests Content-Length headers mismatch",
+        )
+
+
+class TestH1ChunkedNonCacheable(tester.TempestaTest, CommonUtils):
+    """
+    Non-cacheable HTTP1 responses must be forwarded without changes.
+    """
+
+    clients = [{"id": "client", "type": "deproxy", "addr": "${tempesta_ip}", "port": "80"}]
+    backends = [
+        {
+            "id": "backend",
+            "type": "deproxy",
+            "port": "8000",
+            "response": "static",
+            "response_content": "HTTP/1.1 200 OK\r\n"
+            "Content-type: text/html\r\n"
+            f"Last-Modified: {DATE}\r\n"
+            f"Date: {DATE}\r\n"
+            "Server: Deproxy Server\r\n"
+            "Transfer-Encoding: chunked\r\n\r\n"
+            "the body does not actually matter",
+        }
+    ]
+    tempesta = {
+        "config": """
+        server ${server_ip}:8000;
+        cache 0;
+        """
+    }
+
+    def test(self):
+        self.start_all()
+
+        client = self.get_client("client")
+        server = self.get_server("backend")
+
+        got_response = self.send_req(client)
+        response = client.responses[-1] if len(client.responses) else None
+
+        self.assertTrue(got_response, "There should be a response")
+        self.assertEqual(
+            response.headers.get("Transfer-Encoding"), "chunked", "Wrong response Transfer-Encoding"
+        )
+
+        # second request
+        got_response = self.send_req(client)
+        response = client.responses[-1] if len(client.responses) else None
+
+        self.assertTrue(got_response, "There should be a response")
+        self.assertEqual(len(server.requests), 1, "The response has to be server from cache")
```

Comment:
If response "Non-cacheable" and cache disabled, we will have `len(server.requests) == 2`, not 1. Fix assert please.